### PR TITLE
serialization chapter clarification

### PR DESCRIPTION
**Pull request by mustafaiman**
I removed some unnecessary information that may confuse people. It would be a very long section if we explained everything around serialization here. So I kept only the part that explains what is done automatically and why they should consider implementing serialization.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,42 +101,32 @@ See [CONFIG.md](CONFIG.md) for details.
...
+- string
+- number
+- boolean
+- null
+- arrays
 
-The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
-
-Regarding arrays in a serializable object, you can use methods like `writeIntArray` if the array is of a primitive type.
+For other complex types (e.g. JS objects), it uses JSON serialization.
```
Comment (Serdaro) [inline README.md]: ..complex types, e.g., JS objects, it uses.....

File: README.md [hunk 2]
```diff
@@ -101,42 +101,32 @@ See [CONFIG.md](CONFIG.md) for details.
...
 
-If you have nested objects, these nested objects also need to be serializable. Register the serializers for nested objects and the method `writeObject` will not have any problem with finding a suitable serializer for and writing/reading the nested object.
+For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not
+have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic
+between the client and the server and the performance would severely drop. Because predicates run on the server side, the server
```
Comment (Serdaro) [inline README.md]: ...the client and server, and the performance....

File: README.md [hunk 3]
```diff
@@ -101,42 +101,32 @@ See [CONFIG.md](CONFIG.md) for details.
...
-    }
-    this.arrayOfCustomObjects = arrayOfCustomObjects;
-    ...
-}
-```
+The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
 
+## Making Objects Serializable
 
+You can define serialization method for your custom JS objects. Any object that implements `IdentifiedDataSerializable` or
```
Comment (Serdaro) [inline README.md]: ...define the serialization method.....

File: README.md [hunk 4]
```diff
@@ -101,42 +101,32 @@ See [CONFIG.md](CONFIG.md) for details.
...
-    this.arrayOfCustomObjects = arrayOfCustomObjects;
-    ...
-}
-```
+The same applies to MapStore. The server should be able to deserialize your objects in order to store them in MapStore.
 
+## Making Objects Serializable
 
+You can define serialization method for your custom JS objects. Any object that implements `IdentifiedDataSerializable` or 
+`Portable` interfaces are serialized accordingly before sending to the server. If you provide Java counterparts of your objects
```
Comment (Serdaro) [inline README.md]: ...interfaces is serialized accordingly...

File: README.md [hunk 5]
```diff
@@ -101,42 +101,32 @@ See [CONFIG.md](CONFIG.md) for details.
...
+on the server side, Hazelcast Server can deserialize your objects when neccessary(Querying object via predicates, storing objects
```
Comment (Serdaro) [inline README.md]: ...when necessary (querying the object via.....

**Comment by mustafaiman** [comment]
closing in favor of https://github.com/hazelcast/hazelcast-nodejs-client/pull/341

**Pull request closed without merging**